src/Lecture1.hs

```diff
@@ -108,7 +131,16 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strListSum :: [String] -> Int 
+strListSum strList = do
+  if null strList then 0
+  else (read (head strList) :: Int) + strListSum (tail strList)
+
+strSum :: String -> Int 
+strSum str = do
+  let nums = words str
+  if null nums then 0
+  else (read (head nums) :: Int) + strListSum (tail nums)
```

Comment:
There's already a standard function `sum` to find the sum of all elements in the list. So you don't need to write your own function 🙂

Also, you can use the standard function `map` to apply `read` to every element of the list and transform `[String]` to `[Int]`, after which you can apply `sum` to find the sum of all numbers.

So it should be possible to implement this function as a nice one-liner 🚀